lightning/restore/restore.go

```diff
@@ -1344,19 +1360,37 @@ func setSessionConcurrencyVars(ctx context.Context, db *sql.DB, dsn config.DBSto
 // DoChecksum do checksum for tables.
 // table should be in <db>.<table>, format.  e.g. foo.bar
 func DoChecksum(ctx context.Context, db *sql.DB, table string) (*RemoteChecksum, error) {
-	ori, err := increaseGCLifeTime(ctx, db)
-	if err != nil {
-		return nil, errors.Trace(err)
+	var err error
+	helper, ok := ctx.Value(&gcLifeTimeKey).(gcLifeTimeHelper)
+	if !ok {
+		return nil, errors.Errorf("No gcLifeTimeHelper found in context, check context initialization")
+	}
+
+	helper.increaseGCLock.Lock()
+	*(helper.runningJobs) += 1
+	if *(helper.runningJobs) == 1 {
+		err = increaseGCLifeTime(ctx, db)
+		if err != nil {
+			helper.increaseGCLock.Unlock()
+			return nil, errors.Trace(err)
```

Comment:
```suggestion
			return nil, err
```
error that returns from `increaseGCLifeTime` already has the stack trace